src/coprocessor/dag/rpn_expr/impl_compare.rs

```diff
@@ -523,8 +530,8 @@ mod tests {
                 TestCaseCmpOp::NullEQ => ScalarFuncSig::NullEQDecimal,
             };
             let output = RpnFnScalarEvaluator::new()
-                .push_param(arg0.map(|v| Decimal::from_f64(v.into_inner()).unwrap()))
-                .push_param(arg1.map(|v| Decimal::from_f64(v.into_inner()).unwrap()))
+                .push_param(arg0.map(|v| f64_to_decimal(&mut ctx, v.into_inner()).unwrap()))
```

Comment:
ditto. maybe from string is better for keeping unit tests pure.

Reply:
Oops, we should refactor all test cases in this file if we use a string here (I think it's ok to keep float).